src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -343,28 +343,29 @@ public void TestCreateNoWindowProperty(bool value)
         }
 
         [Fact]
-        public void TestWorkingDirectoryProperty()
+        public void TestWorkingDirectoryPropertyInChildProcess()
         {
             CreateDefaultProcess();
 
             // check defaults
             Assert.Equal(string.Empty, _process.StartInfo.WorkingDirectory);
 
-            Process p = CreateProcessLong();
-            p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            ProcessStartInfo psi = new ProcessStartInfo();
+            string sourcePath = Directory.GetCurrentDirectory();
 
-            try
-            {
-                p.Start();
-                Assert.Equal(Directory.GetCurrentDirectory(), p.StartInfo.WorkingDirectory);
-            }
-            finally
+            foreach (string newPath in Directory.GetFiles(sourcePath, "*.*", SearchOption.AllDirectories))
+                File.Copy(newPath, newPath.Replace(sourcePath, TestDirectory), true);
+
+            psi.WorkingDirectory = TestDirectory;
+            Process p = CreateProcess((workingDirectory) =>
             {
-                if (!p.HasExited)
-                    p.Kill();
+                Assert.Equal(Directory.GetCurrentDirectory(), workingDirectory);
+                return SuccessExitCode;
+            }, psi.WorkingDirectory, psi);
```

Comment:
What if psi.WorkingDirectory ignored what you had set it to and always returned /foo ? This test would still pass because you are passing it itself in here.

Instead of passing in psi.WorkingDirectory, I would explicitly pass in whatever you have set it to earlier.